### ISSUE TITLE
transparentOption on color tag (element)

### ISSUE THREAD

**Issue by jasontinnin** [Adobe-Muse/MuCowDocs#20]
It appears that this attribute has been removed.  Is this on purpose?

**Comment by JeffryBooher** [comment]
Do you mean the `<color>` parameter tag?

**Comment by jasontinnin** [comment]
Yes, that's what I meant.

**Comment by JeffryBooher** [comment]
I don't see that the `<color>` ever supported a `transparent` attribute.  Do you mean `supportsNoneColor`?

``` json

    "color": {
        "attributes": [
            "formatHexColor", 
            "label", 
            "defaultValue", 
            "name", 
            "rgbColor", 
            "supportsNoneColor",
            "toolTip"
        ],
        "context": ["parameters", "section"]    
    },

```

**Comment by JeffryBooher** [comment]
Ah, it looks like it was never properly documented but it is still supported in Muse.

**Comment by JeffryBooher** [comment]
Yeah, the `transparentOption` attribute was never documented but I will add this soon.

**Comment by jasontinnin** [comment]
OK cool no worries

**Comment by theperrygroup** [comment]
What is this supposed to look like? I can't get it to work.

**Comment by imheeman** [comment]
Please stop sending mail to me. If not I make it spam!

> On Jan 13, 2016, at 8:28 PM, John Perry <notifications@github.com> wrote:
> 
> What is this supposed to look like? I can't get it to work.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Adobe-Muse/MuCowDocs/issues/20#issuecomment-171276772>.
> 

 

	 	
Hee Man
about.me/chaiheebiz
 <https://about.me/chaiheebiz?promo=email_sig>

**Comment by JeffryBooher** [comment]
@imheeman you can unsubscribe by unfollowing this repo--https://github.com/Adobe-Muse/MuCowDocs/watchers or create a mailbox rule to move it to the trash. If you mark it spam then you may inadvertently block all mail from github which you may not want.

**Comment by JeffryBooher** [comment]
The `transparencyOption` allows for the alpha channel of an RGB color value to be controlled through a numeric input field with value stepper in the OOUI; you need two tags: one for the color picker and one for the numeric input for the alpha value; you need to specify `rgbColor=true` so the result is written as `rgba(r,g,b,a)` and the a value comes from the transparent option.  

Here's an example to play with in a mucow:

``` xml
<color name="bgColor3" label="BG Color 3" rgbColor="true" transparentOption="transparency3" supportsNoneColor="true" defaultValue="0, 0, 0"/>
<number name="transparency3" label="BG Transparency" min="0" max="1" step="0.1" snap="0.01" defaultValue="0.6" />
```

`bgColor3` will be replaced with `rgba(0,0,0,0.6)` at code generation time.

**Comment by theperrygroup** [comment]
That's not what I'm getting. What am I doing wrong?

<pre>
<code>&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
&lt;HTMLWidget name=&quot;widget&quot; formatNumber=&quot;4&quot; creator=&quot;Your name&quot; isResizable=&quot;false&quot; defaultWidth=&quot;150&quot; defaultHeight=&quot;100&quot; localization=&quot;none&quot;&gt;
    
    &lt;parameters&gt;
        &lt;color name=&quot;bgColor3&quot; label=&quot;BG Color 3&quot; rgbColor=&quot;true&quot; transparentOption=&quot;transparency3&quot; supportsNoneColor=&quot;true&quot; defaultValue=&quot;0, 0, 0&quot;/&gt;
        &lt;number name=&quot;transparency3&quot; label=&quot;BG Transparency&quot; min=&quot;0&quot; max=&quot;1&quot; step=&quot;0.1&quot; snap=&quot;0.01&quot; defaultValue=&quot;0.6&quot; /&gt;
    &lt;/parameters&gt;
    
    &lt;pageItemHTML&gt;
    &lt;![CDATA[
        &lt;p&gt;{param_bgColor3}&lt;/p&gt;
        &lt;p&gt;{param_transparency3}&lt;/p&gt;
        &lt;p&gt;{param_bgColor3}{param_transparency3}&lt;/p&gt;
    ]]&gt;  
    &lt;/pageItemHTML&gt;
    
&lt;/HTMLWidget&gt;</code>
</pre>

The above code is just outputting:

<pre>
<code>0,0,0
0.6
0,0,0 0.6</code>
</pre>

And none of them enable a transparency dial in when the color parameter is opened in the OOUI

**Comment by imheeman** [comment]
Spam you later
> On Jan 14, 2016, at 3:57 AM, John Perry <notifications@github.com> wrote:
> 
> That's not what I'm getting. What am I doing wrong?
> 
> <?xml version="1.0" encoding="UTF-8"?>
> <HTMLWidget name="widget" formatNumber="4" creator="Your name" isResizable="false" defaultWidth="150" defaultHeight="100" localization="none">
>     
>     <parameters>
>         <color name="bgColor3" label="BG Color 3" rgbColor="true" transparentOption="transparency3" supportsNoneColor="true" defaultValue="0, 0, 0"/>
>         <number name="transparency3" label="BG Transparency" min="0" max="1" step="0.1" snap="0.01" defaultValue="0.6" />
>     </parameters>
>     
>     <pageItemHTML>
>     <![CDATA[
>         <p>{param_bgColor3}</p>
>         <p>{param_transparency3}</p>
>         <p>{param_bgColor3}{param_transparency3}</p>
>     ]]>  
>     </pageItemHTML>
>     
> </HTMLWidget>
> The above code is just outputting:
> 
> 0,0,0
> 0.6
> 0,0,0 0.6
> —
> Reply to this email directly or view it on GitHub <https://github.com/Adobe-Muse/MuCowDocs/issues/20#issuecomment-171414906>.
> 

 

	 	
Hee Man
about.me/chaiheebiz
 <https://about.me/chaiheebiz?promo=email_sig>

**Comment by JeffryBooher** [comment]
right, you need to write it as `rgba({param_bgColor3},{param_transparency3});`

e.g. 

`.myClass { background-color: rgba({param_bgColor3},{param_transparency3}); }`

**Comment by theperrygroup** [comment]
Okay but what is the point of the <code>transparentOption</code>? If you remove that part, you get the same results. In the example below, you get the same results, the only difference is that the example that uses <code>transparentOption</code> uses a little bit more code:

<pre>
<code>&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
&lt;HTMLWidget name=&quot;widget&quot; formatNumber=&quot;4&quot; creator=&quot;Your name&quot; isResizable=&quot;false&quot; defaultWidth=&quot;150&quot; defaultHeight=&quot;100&quot; localization=&quot;none&quot;&gt;
    
    &lt;parameters&gt;
        &lt;color name=&quot;color1&quot; label=&quot;BG Color 3&quot; rgbColor=&quot;true&quot; transparentOption=&quot;trans1&quot; supportsNoneColor=&quot;true&quot; defaultValue=&quot;0, 0, 0&quot;/&gt;
        &lt;number name=&quot;color1&quot; label=&quot;BG Transparency&quot; min=&quot;0&quot; max=&quot;1&quot; step=&quot;0.1&quot; snap=&quot;0.01&quot; defaultValue=&quot;0.6&quot; /&gt;
        
        &lt;color name=&quot;color2&quot; label=&quot;BG Color 3&quot; rgbColor=&quot;true&quot; supportsNoneColor=&quot;true&quot; defaultValue=&quot;0, 0, 0&quot;/&gt;
        &lt;number name=&quot;trans2&quot; label=&quot;BG Transparency&quot; min=&quot;0&quot; max=&quot;1&quot; step=&quot;0.1&quot; snap=&quot;0.01&quot; defaultValue=&quot;0.6&quot; /&gt;
    &lt;/parameters&gt;
    
    &lt;headHTML&gt;
    &lt;![CDATA[
    &lt;style&gt;
    .firstClass {
        background-color: rgba({param_color1},{param_trans1});
    }
    
    .secondClass {
        background-color: rgba({param_color2},{param_trans2});
    }
    &lt;/style&gt;  
    ]]&gt;    
    &lt;/headHTML&gt;
    
&lt;/HTMLWidget&gt;
</code>
</pre>

**Comment by theperrygroup** [comment]
Is there a difference?

**Comment by JeffryBooher** [comment]
Yes, the color picker will set the value for the transparent option when "no" color is selected so you will a 0 for transparency so that the RGBA parameter for transparent is correctly set. In the latter case, you won't get that.
